src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -791,20 +792,19 @@ static int rbd_nbd(int argc, const char *argv[])
 
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
-  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
-			 CODE_ENVIRONMENT_DAEMON,
-			 CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS);
-  g_ceph_context->_conf->set_val_or_die("pid_file", "");
+
+  vector<const char*> temp_args(args);
+  md_config_t().parse_argv(temp_args);
 
```

Comment:
nitpicking: I would prefer if you passed argc, argv to do_map() and removed `env_to_vec(args)`  in rbd_nbd(). So we wouldn't parse unnecessary the environment here and and there would not be unnecessary code modifications (s/args/temp_args/) below.